### PR TITLE
rfc12: add security arch RFC

### PR DESCRIPTION
**Pull request by garlick**
As discussed in flux-framework/flux-core#699, this is a raw draft of a Flux security architecture document.

I think it accurately documents the single-user, non-bootstrap strategy we have now.  The intent here is to do that, so that we have a hopping off point for further design work - not to advance the design just yet.

### CODE REVIEW

File: spell.en.pws [hunk 1]
```diff
@@ -147,3 +147,49 @@ Google
...
+Melnikov
+NaCl
+PMI
+SASL
+SELinux
+UID
+et
+al
+cgroups
+evesdropping
```
Comment (SteVwonder) [inline spell.en.pws]: This should be "eavesdropping"

**Comment by garlick** [comment]
Oh thank you!

**Comment by grondo** [comment]
This is really nicely done. I'm ready to merge if you are. 

I did happen to notice that `libk` snuck into the dictionary but doesn't seem to be necessary.

**Comment by grondo** [comment]
> I did happen to notice that libk snuck into the dictionary but doesn't seem to be necessary.

oh, maybe because of `libk5crypto`?  Either way I don't think it matters too much.

**Comment by garlick** [comment]
Thanks!  I don't think it hurts to merge this.  It's just a starting point.

**Comment by garlick** [comment]
Let me squash this down to one commit though.

**Pull request merged**